### PR TITLE
compatible: Add pytest-operator plugin for test grouping

### PR DESCRIPTION
**Pull request by carlcsaposs-canonical**
Will be used in future integration test reusable workflow

### CODE REVIEW

File: python/pytest_plugins/pytest_operator_groups/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
+Run integration tests on parallel GitHub runners
+
+By default, [pytest-operator](https://github.com/charmed-kubernetes/pytest-operator) runs each test file (Python module) in a separate Juju model. Usually, on GitHub, each integration test file is run in parallel (on separate runners).
+
+This plugin allows further parallelization—an individual test file can be split into multiple groups. Each group can run on a separate GitHub runner.
```
Comment (taurus-forever) [inline python/pytest_plugins/pytest_operator_groups/README.md:5]: Nice!

**Review by taurus-forever — APPROVED** [review]
Very promising!

File: python/pytest_plugins/pytest_operator_groups/README.md (hunk 1 above)
Comment (paulomach) [reply python/pytest_plugins/pytest_operator_groups/README.md:5]: super nice!

File: python/pytest_plugins/pytest_operator_groups/pytest_operator_groups/plugin.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        # Remove tests that do not match the selected group number
+        filtered_items = []
+        for function in items:
+            group_number = _get_group_number(function)
+            if group_number is None:
+                function.add_marker(pytest.mark.skip("Missing group number"))
+                filtered_items.append(function)
+            elif group_number == config.option.group:
+                filtered_items.append(function)
+        items[:] = filtered_items
```
Comment (paulomach) [inline python/pytest_plugins/pytest_operator_groups/pytest_operator_groups/plugin.py:93]: curiosity: will not this just override items content? why not just
```suggestion
        items = filtered_items
```

**Review by paulomach — APPROVED** [review]

**Pull request merged**